### DISCUSSION TITLE
Target file without the version tag

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-pde/eclipse.pde#405] [Q&A]
Today I "discovered" that I have leave out the version number for a plug-in in the target file.

Is not using the version the same as using version="0.0.0"?
````
 <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
		 <repository location="https://download.eclipse.org/technology/babel/update-site/latest/"/>
		 <unit id="org.eclipse.babel.nls_eclipse_de.feature.group" version="0.0.0"/>
	 </location>

````

````
 <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
		 <repository location="https://download.eclipse.org/technology/babel/update-site/latest/"/>
		 <unit id="org.eclipse.babel.nls_eclipse_de.feature.group"/>
	 </location>

````

**Comment by laeubi** [comment]
> Today I "discovered" that I have leave out the version number for a plug-in in the target file.

And what has happened?

> Is not using the version the same as using version="0.0.0"?

"0.0.0" is a special version (the `emptyVersion`) still it is not the same as `""` or `null` (whatever will be the outcome of a missing attribute in the xml). Also the UI actually never offers you that (neither `0.0.0` nor not specify any version at all) so at least this is most likely unspecified behavior.

But as the target location is responsible for parsing the xml it could still interpret an empty string or an missing attribute as the `emptyVersion` you should just be aware that there are more tools / code parsing such file and you need to test your whole tool chain how they behave here.

So to summarize, it might be more valuable to not "forget" the version, or even add UI to support "using highest version from repository" so one has no need to edit that file manually (and thus can't forget about it).

**Reply by vogella** [reply]
> > Today I "discovered" that I have leave out the version number for a plug-in in the target file.
> 
> And what has happened?

Seems to work fine / the same in PDE. 

I typical like having to maintain less data but if some tools may require this, I will continue to maintain it.

Thanks for the answer.
